sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -395,12 +396,19 @@ class Analyzer(
         Project(projectList = expanded, p.child)
       // If the aggregate function argument contains Stars, expand it.
       case a: Aggregate if containsStar(a.aggregateExpressions) =>
-        val expanded = a.aggregateExpressions.flatMap {
-          case s: Star => s.expand(a.child, resolver)
-          case o if containsStar(o :: Nil) => expandStarExpression(o, a.child) :: Nil
-          case o => o :: Nil
-        }.map(_.asInstanceOf[NamedExpression])
-        a.copy(aggregateExpressions = expanded)
+        if (conf.groupByOrdinal && a.groupingExpressions.exists(IntegerIndex.unapply(_).nonEmpty)) {
+          failAnalysis(
+            "Group by position: star is not allowed to use in the select list " +
+              "when using ordinals in group by")
+        } else {
+          val expanded = a.aggregateExpressions.flatMap {
+            case s: Star => s.expand(a.child, resolver)
+            case u @ UnresolvedAlias(_: Star, _) => expandStarExpression(u.child, a.child) :: Nil
```

Comment:
I think this is intentionally added by `CatalystQl`. I can double check if this is the root cause.

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/CatalystQl.scala#L224-L225